app/javascript/components/editor/header/usePanel.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { useEffect, useState, useRef } from 'react'
```

Comment:
This hook allows us to use show / hide a panel when clicking on a specific button. This also closes the panel when the user clicks outside it.

Reply:
Excellent work!